### ISSUE TITLE
5.0 RC4 - Scalar mappings are storing Decimal as double

### ISSUE THREAD

**Issue by niemyjski** [elastic/elasticsearch-net#2463]
I noticed an issue where you can't round trip  `Decimal.MaxValue` as the mapping type is double. This occurs due to an overflow (`Value was either too large or too small for a Decimal.`). It might be good to have unit tests for all the c# numeric types and do max value of each type and then call AutoMap / Scalar and then attempt to round trip them. As precision will be lost or gained and it should be handled :).

The following snippet shows off what's happening in the simplest of terms (http://stackoverflow.com/questions/21337241/convert-from-decimal-to-single-and-back-why-does-it-throw-an-overflow-exception).
```cs
double dbl = (double)Decimal.MaxValue;
decimal dec = (decimal)dbl;
```

I think this has to due with two things that should be noted. The types supported by elastic don't always match up to .NET Types and serialization.

**Comment by russcam** [comment]
related https://github.com/elastic/elasticsearch/issues/17006

**Comment by russcam** [comment]
We discussed earlier this week about what we can or should do regarding this. 

We decided that we would document the usage of `decimal` POCO property types in the automapping documentation (I think a separate section on types would be better in the future; I'll add an item to #2248) and how these map to Elasticsearch's `double` type. Attempting to handle an overflow exception in the client seems disingenuous, even if it originates from a `Decimal.MinValue` or `Decimal.MaxValue` conversion to `double`.

Hope that makes sense.
